### PR TITLE
warn/xfd: force chosen select menu to display over the dialog; padding fix in xfd

### PR DESCRIPTION
**Pull request by siddharthvp**
Removes the need to set a max-height to it. Also enables more templates to be viewed without having to scroll.

Code credits: https://github.com/harvesthq/chosen/issues/1390#issuecomment-21397245

Would have added this to #662 hadn't it been merged by now.

### CODE REVIEW

File: modules/twinklewarn.js [hunk 1]
```diff
@@ -1165,8 +1165,8 @@ Twinkle.warn.callback.change_category = function twinklewarnCallbackChangeCatego
 			.chosen({width: '100%', search_contains: true})
 			.change(Twinkle.warn.callback.change_subcategory);
 
-		// Limit the max height of select dropdown to prevent dialog box from becoming scrollable
-		$('.chosen-results').css({'overflow': 'auto', 'max-height': '180px'});
+		// Force chosen select menu to display over the dialog when it overflows it
```
Comment (Amorymeltzer) [inline modules/twinklewarn.js]:
```suggestion
		// Force chosen select menu to display over the dialog when it overflows it
		// https://github.com/harvesthq/chosen/issues/1390#issuecomment-21397245
```

File: modules/twinklewarn.js [hunk 2]
```diff
@@ -1165,8 +1165,8 @@ Twinkle.warn.callback.change_category = function twinklewarnCallbackChangeCatego
...
+		mw.util.addCSS('.ui-dialog .ui-dialog-content { overflow:visible !important;} .ui-dialog { overflow: inherit !important; }')
```
Comment (Amorymeltzer) [inline modules/twinklewarn.js]:
```suggestion
		mw.util.addCSS('.ui-dialog.morebits-dialog .morebits-dialog-content { overflow:visible !important;} .ui-dialog.morebits-dialog { overflow: inherit !important; }');
```

**Comment by Amorymeltzer** [comment]
Merge it fast, you say!  Wait, there's more, you say!  Sheesh, can't win! 😛 

Anyway, @siddharthvp, shouldn't this kind of thing be in `morebits.css`?  Shouldn't that reduce the need for the inherit?  That may have unintended consequences?  I *think* `overflow: hidden` is from mainly from the skin interface, although d9e6e52 from @MusikAnimal in 2016 (#280) might suggest otherwise?  I can't trigger that issue, so perhaps things have changed upstream?

**Comment by siddharthvp** [comment]
> Merge it fast, you say! Wait, there's more, you say! Sheesh, can't win! 😛

lol, technically i had asked you to make them live, not merge (before the pref option was added). 

> Anyway, @siddharthvp, shouldn't this kind of thing be in morebits.css? Shouldn't that reduce the need for the inherit? That may have unintended consequences? I think overflow: hidden is from mainly from the skin interface, although d9e6e52 from @MusikAnimal in 2016 (#280) might suggest otherwise? I can't trigger that issue, so perhaps things have changed upstream?

meh, now you are analysing this? I was just excited to see that guy's code worked and copy-pasted it blindly. I have no idea what overflow property `ui.dialog` is inheriting and from where. If moving to `morebits.css` won't cause any side-effects, sure we'll move it.

**Comment by Amorymeltzer** [comment]
@siddharthvp I agree this should go out soon, so let's do this: I'll merge this (done) and push it on-wiki (done) then I'll play around with morebits.css on testwiki when I get a moment and move 'em if everything seems copacetic.

**Pull request merged**